File: 0890-lemonade-change/0890-lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        li=[]
        res=True
        for i in bills:
            if i==5:
                li.append(i)
            if i==10:
                if 5 in li:
                    li.append(i)
                    li.remove(5)
                else:
                    res=False
                    break
            if i==20:
                if 5 in li and 10 in li:
                    li.append(i)
                    li.remove(5)
                    li.remove(10)
                elif 5 in li and li.count(5)>2:
                    li.append(i)
                    li.remove(5)
                    li.remove(5)
                    li.remove(5)
                else:
                    res=False
                    break
        return res

File: 0890-lemonade-change/0890-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int c5=0,c10=0;
        bool res=true;
        for(int i:bills){
            if(i==5){
                c5++;
            }
            if(i==10){
                if(c5>0){
                    c10++;
                    c5--;
                }
                else{
                    res=false;
                    break;
                }
            }
            if(i==20){
                if(c5>0 && c10>0){
                    c5--;
                    c10--;
                }
                else if(c5>2){
                    c5-=3;
                }
                else{
                    res=false;
                    break;
                }
            }
        }
        return res;
    }
};